server/gamemodes/chess.py

```diff
@@ -325,27 +290,29 @@ def get_king_moves(self, king: models.Piece) -> typing.Iterator[int, int]:
                     continue
                 rank = king.rank + rank_direction
                 file = king.file + file_direction
+                if not self.on_board(file, rank):
+                    continue
                 victim = self.get_piece(rank, file)
-                if (not victim) or (victim.side != king.side):
+                if (((not victim) or (victim.side != king.side))
+                        and not self.hypothetical_check(king.side,
+                        (king, rank, file))):
```

Comment:
```suggestion
                if (not victim) or (victim.side != king.side):
```
`hypothetical_check` is already called by `possible_moves`.

Reply:
Ok, but is that the only time you will call that function?  This is a useful UI function too, since when you click a piece it can show where the possible moves are.  No worries though, we can just create a wrapper function possible_moves_piece which checks for check as well.

Reply:
I'm not sure what you mean? `possible_moves` determines the type of piece, then calls the relevant `get_<piece>_moves` function, then tests each result with `hypothetical_check`. Therefore, there is no need for the `get_<piece>_moves` functions to call `hypothetical_check` themselves.

Reply:
Sorry, I meant to change this too.  What I meant was that `possible_moves` returns all possible moves for a game rather than just for one piece.  I guess this is ok too, though it increases overhead.  Especially in the case where the UI wants to ask, what are the possible moves for a single selected piece, so that it can display them for the user (like the target circles in poly).